src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -255,6 +256,13 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
     ReferenceCountedChannel execChannel = null;
     ReferenceCountedChannel cacheChannel = null;
     ReferenceCountedChannel downloaderChannel = null;
+
+    int poolSize = 1;
+    BuildRequestOptions buildRequestOptions = env.getOptions().getOptions(BuildRequestOptions.class);
+    if (buildRequestOptions != null) {
+      // TODO(chiwang): Should we use a more intelligent way to calculate poolSize?
```

Comment:
yeah, this is my main question: How did you come up with this number?

Reply:
I believe its based on the recommendation here - https://github.com/bazelbuild/bazel/issues/11801#issuecomment-672973245

Reply:
That makes sense indeed. Chi, can you please document this in the code?

Reply:
Note that the recommendation there is to either take into account the number of IPs (the number of connections, aka subchannels, per channel), or to assume 2 IPs per host which would be jobs/100 "Channels" in this pool.

Looking at this code, just assuming 2 makes more sense to me than having to also reimplement the subchannel->channel interface, which I think would be required to count the IPs. So I vote for simply changing this constant to 100 and documenting clearly why it is what it is.